File: RxCommLib/src/main/java/info/mx/comlib/retrofit/service/model/google/Geometry.java

package info.mx.comlib.retrofit.service.model.google;

import android.os.Parcel;
import android.os.Parcelable;

import com.google.gson.annotations.Expose;
import com.google.gson.annotations.SerializedName;

import java.io.Serializable;

@SuppressWarnings("JavadocDeclaration")
public class Geometry implements Serializable, Parcelable {

    @SerializedName("bounds")
    @Expose
    private Bounds bounds;
    @SerializedName("location")
    @Expose
    private Location location;
    @SerializedName("location_type")
    @Expose
    private String locationType;
    @SerializedName("viewport")
    @Expose
    private Viewport viewport;
    public final static Creator<Geometry> CREATOR = new Creator<>() {

        public Geometry createFromParcel(Parcel in) {
            return new Geometry(in);
        }

        public Geometry[] newArray(int size) {
            return (new Geometry[size]);
        }

    };
    private final static long serialVersionUID = -5974024649684848204L;

    protected Geometry(Parcel in) {
        this.bounds = ((Bounds) in.readValue((Bounds.class.getClassLoader())));
        this.location = ((Location) in.readValue((Location.class.getClassLoader())));
        this.locationType = ((String) in.readValue((String.class.getClassLoader())));
        this.viewport = ((Viewport) in.readValue((Viewport.class.getClassLoader())));
    }

    /**
     * No args constructor for use in serialization
     */
    public Geometry() {
    }

    /**
     * @param bounds
     * @param viewport
     * @param location
     * @param locationType
     */
    public Geometry(Bounds bounds, Location location, String locationType, Viewport viewport) {
        super();
        this.bounds = bounds;
        this.location = location;
        this.locationType = locationType;
        this.viewport = viewport;
    }

    public Bounds getBounds() {
        return bounds;
    }

    public void setBounds(Bounds bounds) {
        this.bounds = bounds;
    }

    public Location getLocation() {
        return location;
    }

    public void setLocation(Location location) {
        this.location = location;
    }

    public String getLocationType() {
        return locationType;
    }

    public void setLocationType(String locationType) {
        this.locationType = locationType;
    }

    public Viewport getViewport() {
        return viewport;
    }

    public void setViewport(Viewport viewport) {
        this.viewport = viewport;
    }

    public void writeToParcel(Parcel dest, int flags) {
        dest.writeValue(bounds);
        dest.writeValue(location);
        dest.writeValue(locationType);
        dest.writeValue(viewport);
    }

    public int describeContents() {
        return 0;
    }

}

File: RxCommLib/src/main/java/info/mx/comlib/retrofit/service/model/google/Geometry.kt
package info.mx.comlib.retrofit.service.model.google

import android.os.Parcel
import android.os.Parcelable
import com.google.gson.annotations.Expose
import com.google.gson.annotations.SerializedName
import java.io.Serializable

open class Geometry : Serializable, Parcelable {
    @SerializedName("bounds")
    @Expose
    var bounds: Bounds? = null

    @SerializedName("location")
    @Expose
    var location: Location? = null

    @SerializedName("location_type")
    @Expose
    var locationType: String? = null

    @SerializedName("viewport")
    @Expose
    var viewport: Viewport? = null

    protected constructor(`in`: Parcel) {
        this.bounds = (`in`.readValue((Bounds::class.java.classLoader)) as Bounds?)
        this.location = (`in`.readValue((Location::class.java.classLoader)) as Location?)
        this.locationType = (`in`.readValue((String::class.java.classLoader)) as String?)
        this.viewport = (`in`.readValue((Viewport::class.java.classLoader)) as Viewport?)
    }

    override fun writeToParcel(dest: Parcel, flags: Int) {
        dest.writeValue(bounds)
        dest.writeValue(location)
        dest.writeValue(locationType)
        dest.writeValue(viewport)
    }

    override fun describeContents(): Int {
        return 0
    }

    companion object {
        @JvmField
        val CREATOR: Parcelable.Creator<Geometry?> = object : Parcelable.Creator<Geometry?> {
            override fun createFromParcel(`in`: Parcel): Geometry {
                return Geometry(`in`)
            }

            override fun newArray(size: Int): Array<Geometry?> {
                return arrayOfNulls(size)
            }
        }
    }
}
